### DISCUSSION TITLE
Diff of output from LLM instead of full code

### DISCUSSION THREAD

**Discussion by psalzman** [cline/cline#424] [Feature Requests]
Couldn't the LLM be instructed to send a diff of the updated as opposed to the contents of the entire file?  Wouldn't that cut back on the 'keep existing functions' problem that so many are having?

**Comment by saoudrizwan** [comment]
Cline v3 now has diff editing! When Cline needs to make targeted edits to large files, he now uses a search & replace diff format where he only has to output the content to replace, rather than the whole file. This allows for faster, more reliable edits and prevents code deletion issues where previously, unmodified sections of code might have been replaced with comments like // rest of code here. Cline can also fall back to whole file editing for when it makes sense, e.g. replacing boilerplate files, complete refactors, or when diff editing fails.
